.github/workflows/release.yml

```
@@ -0,0 +1,74 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+
+  publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      # write is needed for:
+      # - OIDC for cosign's use in ecm-distro-tools/publish-image.
+      # - Read vault secrets in rancher-eio/read-vault-secrets.
+      id-token: write
+    
+    strategy:
+      matrix:
+        include:
+          - platforms: linux/amd64,linux/arm64
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+    - name: Read secrets
+      env:
+        DOCKER_REPO: ${{ secrets.DOCKER_REPO }}
+        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+        PRIME_REGISTRY: ${{ secrets.PRIME_REGISTRY }}
+        PRIME_REPO: ${{ secrets.PRIME_REPO }}
+        PRIME_REGISTRY_USERNAME: ${{ secrets.PRIME_REGISTRY_USERNAME }}
+        PRIME_REGISTRY_PASSWORD: ${{ secrets.PRIME_REGISTRY_PASSWORD }}
+      run: |
+        echo "DOCKER_REPO=$DOCKER_REPO" >> $GITHUB_ENV
+        echo "DOCKER_USERNAME=$DOCKER_USERNAME" >> $GITHUB_ENV
+        echo "DOCKER_PASSWORD=$DOCKER_PASSWORD" >> $GITHUB_ENV
+        echo "PRIME_REGISTRY=$PRIME_REGISTRY" >> $GITHUB_ENV
+        echo "PRIME_REPO=$PRIME_REPO" >> $GITHUB_ENV
+        echo "PRIME_REGISTRY_USERNAME=$PRIME_REGISTRY_USERNAME" >> $GITHUB_ENV
+        echo "PRIME_REGISTRY_PASSWORD=$PRIME_REGISTRY_PASSWORD" >> $GITHUB_ENV
+
+    # TODO: Read secrets from vault
+
+    - name: Publish manifest
+      uses: rancher/ecm-distro-tools/actions/publish-image@master
+      with:
+        image: prometheus-exporter
+        tag: ${{ github.ref_name }}${{ matrix.tag-suffix }}
```

If publishing only multi-arch images:

```suggestion
        tag: ${{ github.ref_name }}
```